### DISCUSSION TITLE
Ability to turn off sound

### DISCUSSION THREAD

**Discussion by endolith** [streetcomplete/StreetComplete#5039] [Ideas]
I'd like an option to turn off the pop sounds

**Comment by FloEdelmann** [comment]
Please see <https://wiki.openstreetmap.org/wiki/StreetComplete/FAQ#How_can_I_disable/enable_sound_effects?>.

**Reply by endolith** [reply]
That just links to https://developer.android.com/reference/android/provider/Settings.System#SOUND_EFFECTS_ENABLED  Is that meant for end-users or developers?  I don't understand how to disable the sound effects from that.  There should just be a checkbox in the app's settings.

**Reply by FloEdelmann** [reply]
The app uses sound effects if sound effects are enabled in the whole Android system. You can turn them off in the Android settings (different exact location for each manufacturer or Android version).

**Reply by endolith** [reply]
I don't want to turn them off for the entire operating system, just a checkbox in the settings for this particular app.

**Reply by mnalis** [reply]
- See https://github.com/streetcomplete/StreetComplete/issues/1187,https://github.com/streetcomplete/StreetComplete/issues/1167, https://github.com/streetcomplete/StreetComplete/issues/2966 and https://github.com/streetcomplete/StreetComplete/issues/3539; it was suggested before but rejected. StreetComplete seems to take a principle to **minimize amount of user preferences**, and only add them if there is no other way.

You can try:
- disabling it for entire Android as already suggested. Maybe you'll like it without missing anything (I do!)
- if not, you can try to make your case. If you decide to go that way, before creating new issue, please first read all related issues, and try to explain exactly (with examples!) why disabling such sounds globally is not a workable solution for you. Perhaps find others with other use cases if you can and mention that too. Here is my example which finally persuaded project to add per-app language chooser (while originally SC only followed system language) - https://github.com/streetcomplete/StreetComplete/issues/2643#issuecomment-792952040 (you'll note that it took some effort to collect and present the various reasons in persuadable way)
- or you can try suggesting the feature to [SCEE](https://wiki.openstreetmap.org/wiki/SCEE) fork, which doesn't have an issue with a lots of preferences so might implement it more easily; and then switch to that fork of the app.

**Reply by mnalis** [reply]
Perhaps I should clarify, [that setting](https://github.com/streetcomplete/StreetComplete/issues/2966#issuecomment-862716483) **does not disable all sounds** on the phone. In fact, by disabling it, you may not even notice that sound is disabled in anything else, or it may disable just some equally annoying clicks and redundant noises elsewhere. It has **no effect on regular sounds** (like ringer, alarms, phone audio, music output, game sounds, ...).

It should've really been named `ANNOYING_SOUNDS_ENABLED`. I've never even had a phone with that setting enabled, and when I tried to enable to hear those sounds, I've quickly hurried to disable it again. Just disable it and keep it disabled; you won't be missing it.

**Reply by endolith** [reply]
It's a setting that applies to all apps on the phone, no?

**Reply by mnalis** [reply]
Only to apps that select to use that functionality (which, to my experience, usually varies between _"very little"_ or _"none at all"_, depending on your exact app list).

**Reply by endolith** [reply]
OK, but I don't want to affect all apps, I only want to turn off sounds for StreetComplete.

**Reply by mnalis** [reply]
I understand clearly what you want @endolith. But it is not in **my** power to give it. I've just tried to be helpful by explaining why this feature was not implemented previously (by giving links to previous issues), and given few options available to you in the [post above](https://github.com/streetcomplete/StreetComplete/discussions/5039#discussioncomment-5979217) - which you might (or might not) want to try.
